src/server/game/Entities/Object/Object.h

```diff
@@ -977,7 +977,7 @@ class WorldObject : public Object, public WorldLocation
 
         bool isActiveObject() const { return m_isActive; }
         void setActive(bool isActiveObject);
-        bool IsVisibilityOverridden() const { return m_isVisibilityDistanceOverride || m_isActive; }
```

Comment:
why? you're using setActive() so you can keep it

Reply:
No need to increase visibility distance for active creatures. This should be controlled solely via "isLarge" flag in creature_addon or creature_template_addon.